test/integration/hazelcast_webhook_test.go

```diff
@@ -104,4 +116,30 @@ var _ = Describe("Hazelcast webhook", func() {
 				Should(MatchError(ContainSubstring("when exposeExternally.type is set to \"Unisocket\", exposeExternally.memberAccess must not be set")))
 		})
 	})
+
+	Context("Hazelcast HighAvailabilityMode", func() {
+		It("should fail to update", Label("fast"), func() {
+			zoneHASpec := test.HazelcastSpec(defaultSpecValues, ee)
+			zoneHASpec.HighAvailabilityMode = "ZONE"
+
+			hz := &hazelcastv1alpha1.Hazelcast{
+				ObjectMeta: GetRandomObjectMetaWithAnnotation(&zoneHASpec),
+				Spec:       zoneHASpec,
+			}
+
+			Expect(k8sClient.Create(context.Background(), hz)).Should(Succeed())
+			test.CheckHazelcastCR(hz, defaultSpecValues, ee)
+
+			hz.Spec.HighAvailabilityMode = hazelcastv1alpha1.HighAvailabilityNodeMode
+
+			err := k8sClient.Update(context.Background(), hz)
+			if err != nil && !errors.IsConflict(err) {
+				Expect(err).
+					Should(MatchError(ContainSubstring("highAvailabilityMode cannot be updated")))
+			}
```

Comment:
```suggestion
			var err error
			for {
				Expect(k8sClient.Get(
					context.Background(), types.NamespacedName{Namespace: hz.Namespace, Name: hz.Name}, hz)).Should(Succeed())
				hz.Spec.HighAvailabilityMode = hazelcastv1alpha1.HighAvailabilityNodeMode
				err = k8sClient.Update(context.Background(), hz)
				if errors.IsConflict(err) {
					continue
				}
				break
			}
			Expect(err).
				Should(MatchError(ContainSubstring("highAvailabilityMode cannot be updated")))
```
I would suggest the following changes that retry the update in case of the conflict; otherwise, in case of the conflict, the test will test nothing.

Reply:
good catch 🤦‍♂️ 😄